contracts/UsingHelpers.sol

```diff
@@ -2,111 +2,100 @@
+    * @param _status current status of a voucher.
+    */
+function isStateRedemptionSigned(uint8 _status)
+    pure
+    returns (bool)
+{
+    return _status == determineStatus(determineStatus(0, VoucherState.COMMIT), VoucherState.REDEEM);
+}
 
-    /**
-     * @notice Based on its lifecycle, voucher can have many different statuses. Checks whether a voucher is in Refunded state.
-     * @param _status current status of a voucher.
-     */
-    function isStateRefunded(uint8 _status) internal pure returns (bool) {
-        return _status == determineStatus(determineStatus(0, IDX_COMMIT), IDX_REFUND);
-    }
+/**
+    * @notice Based on its lifecycle, voucher can have many different statuses. Checks whether a voucher is in Refunded state.
+    * @param _status current status of a voucher.
+    */
+function isStateRefunded(uint8 _status) pure returns (bool) {
+    return _status == determineStatus(determineStatus(0, VoucherState.COMMIT), VoucherState.REFUND);
+}
 
-    /**
-     * @notice Based on its lifecycle, voucher can have many different statuses. Checks whether a voucher is in Expired state.
-     * @param _status current status of a voucher.
-     */
-    function isStateExpired(uint8 _status) internal pure returns (bool) {
-        return _status == determineStatus(determineStatus(0, IDX_COMMIT), IDX_EXPIRE);
-    }
+/**
+    * @notice Based on its lifecycle, voucher can have many different statuses. Checks whether a voucher is in Expired state.
+    * @param _status current status of a voucher.
+    */
+function isStateExpired(uint8 _status) pure returns (bool) {
+    return _status == determineStatus(determineStatus(0, VoucherState.COMMIT), VoucherState.EXPIRE);
+}
 
-    /**
-     * @notice Based on its lifecycle, voucher can have many different statuses. Checks the current status a voucher is at.
-     * @param _status current status of a voucher.
-     * @param _idx status to compare.
-     */
-    function isStatus(uint8 _status, uint8 _idx) internal pure returns (bool) {
-        return (_status >> _idx) & ONE == 1;
-    }
+/**
+    * @notice Based on its lifecycle, voucher can have many different statuses. Checks the current status a voucher is at.
+    * @param _status current status of a voucher.
+    * @param _idx status to compare.
+    */
+function isStatus(uint8 _status, VoucherState _idx) pure returns (bool) {
+    return (_status >> uint8(_idx)) & ONE == 1;
+}
 
-    /**
-     * @notice Set voucher status.
-     * @param _status previous status.
-     * @param _changeIdx next status.
-     */
-    function determineStatus(uint8 _status, uint8 _changeIdx)
-        internal
-        pure
-        returns (uint8)
-    {
-        return _status | (ONE << _changeIdx);
-    }
+/**
+    * @notice Set voucher status.
+    * @param _status previous status.
+    * @param _changeIdx next status.
+    */
+function determineStatus(uint8 _status, VoucherState _changeIdx)
```

Comment:
Shouldn't _status also be of type VoucherStatus?

Reply:
The same explanation as above.

Reply:
OK